src/mix.cr

```diff
@@ -0,0 +1,255 @@
+require "./lib_mix"
+require "./sdl"
+
+module SDL
+  module MIX
+    alias Init = LibMIX::Init
+    MAX_VOLUME = LibMIX::MIN_MAX_VOLUME
+    
+    enum Type
+      AIFF
+      FLAC
+      MIDI
+      MOD
+      MP3
+      OGG
+      VOC
+      WAV
+    end
+
+    # Used to load the support for the flags. `quit` must be called during app cleanup
+    def self.init(flags : Init)
+      ret = LibMIX.init(flags)
+      unless (ret & flags.value) == flags.value
+        raise SDL::Error.new("Mix_Init failed to init #{flags}")
+      end
+    end
+
+    def self.quit
+      LibMIX.quit
+    end
+
+    # This is required to initial SDL_Mixer. It must be called before using any other function, but AFTER SDL has been initialized.
+    def self.open(freq = 44100, format = 0_u16, channels = 2, sample_size = 2048)
+      format = format == 0 ? LibMIX::MIX_DEFAULT_FORMAT : format.to_u16
+      ret = LibMIX.open_audio(freq, format, channels, sample_size)
+      raise SDL::Error.new("Mix_OpenAudio") unless ret == 0
+      ret
+    end
+
+    def self.query_spec(freq = 44100, format = 0_u16, channels = 2)
+      audio_open_count = LibMIX.query_spec(freq, format, channels)
+      raise SDL::Error.new("Mix_QuerySpec: #{LibMIX.get_error}") if audio_open_count < 1
+      audio_open_count
+    end
+
+    def self.close
+      LibMIX.close_audio
+    end
+
+  end
+
+  module MIX
+    class Sample
+      protected getter sample
```

Comment:
You may drop the getter, use `@sample` and expose it as `#to_unsafe` only, so you may pass `self` to C bindings. This would follow what the design of other SDL wrapper classes.